Redacted comment html for deleted and hidden comments

allouis opened the pull request:
refs https://github.com/TryGhost/Team/issues/1745

I've done this at the model layer, which works for now, but isn't great
if we eventually create some kind of Admin section, in which case we'll
want to expose the html of the comment, regardless of status. This is
something we can address as/when we get to it.

allouis committed:
Redacted comment html for deleted and hidden comments

refs https://github.com/TryGhost/Team/issues/1745

I've done this at the model layer, which works for now, but isn't great
if we eventually create some kind of Admin section, in which case we'll
want to expose the html of the comment, regardless of status. This is
something we can address as/when we get to it.

allouis commented:
@ErisDS @SimonBackx What do you think of redacting this in the model layer, as opposed to the API serialiser layer (in the mapper)

ErisDS commented:
Curious what the reason is for doing it in the model layer rather than the mapper?

As I understand, we're trying to move away from having anything other than immutable business logic in the models...  (e.g. redact password digest from user model cos no API should output it) 

I have concerns that this could inadvertently lead to us setting the html to null in the admin API for comments when we add one... although they're not at all well founded. 

To me this feels specific to the API?

allouis committed:
Switched to redacting html in the mapper

allouis committed:
fixup! Switched to redacting html in the mapper

allouis committed:
Updated mapper to only redact for the members API

allouis commented:
After discussion offline, I'm updating this PR to redact the html contents in the mapper

allouis committed:
Added validation to ensure cannot edit deleted comments

This allows us to send back `null` as the html field, rather than
removing the field completely, as we are not at risk of losing data if
we send back the comment to the API, overwriting the html property.

allouis committed:
fix lint

allouis committed:
Fixed error message for editing comments

allouis committed:
Merge branch 'main' into redact-comments
